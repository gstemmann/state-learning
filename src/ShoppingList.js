import React, { useState } from 'react';
import { v4 as uuid } from 'uuid';
import SingleItem from './SingleItem';
import FormForNewItem from './FormForNewItem';

const ShoppingList = () => {
    const INITIAL_STATE = [{
        id: uuid(),
        name: 'blue butter',
        qty: 'while you wait'
    },
    {
        id: uuid(), 
        name: 'red butter',
        qty: 'while you wait'
    }];

    const [currentItems, changeItems] = useState([INITIAL_STATE]);
    const addItemToList = (brandNewItem) => {
        changeItems(currentItems => [ ...currentItems, { ...brandNewItem, id:uuid() } ])
    }
   
    return (
        <div>
            <h1>Shopping List</h1>
            <FormForNewItem addItemToList={addItemToList} />
            <ul>
                {currentItems.map(({ id, name, qty}) => (
                    <li>
                        <SingleItem id={id} name={name} qty={qty} key={id} />
                    </li>
                ))}
            </ul>
        </div>
    )
}

export default ShoppingList;