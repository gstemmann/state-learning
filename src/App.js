// import logo from './logo.svg';
// import Circle from "./Circle";
// import ColoredCircles from "./ColoredCircles";
// // import Button from './button';
import Counter from './Counter';
import ShoppingList from './ShoppingList';
// // import Group from './Group';
// import NumbersList from './NumbersList';
// import UserForm from './UserForm';
// import ShoppingList from './ShoppingList';
import Toggler from './Toggler';
import SimpleForm from './SimpleForm';



function App() {
  return (
    <div className="App">
    <SimpleForm />
       
    </div>
  );
}

export default App;
