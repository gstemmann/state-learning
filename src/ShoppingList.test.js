import React from "react";
import { render, screen, fireEvent } from '@testing-library/react';
import App from './App';
import ShoppingList from './ShoppingList';



it('renders the ShoppingList component', () => {
  render(<ShoppingList />);
});
it('matches snapshot', function() {
    const {asFragment} = render(<ShoppingList />);
    expect(asFragment()).toMatchSnapshot();
  });


test('form adds new item', function() {
  render(<ShoppingList />);
  expect(screen.queryByText('purple butter')).not.toBeInTheDocument();
  const button = screen.queryByText('Add Item');
  const input = screen.getByLabelText('Product Name:');
  fireEvent.change(input, { target: { value: 'purple butter' } });
  expect(screen.queryByText('purple butter')).not.toBeInTheDocument();
  fireEvent.click(button);
  expect(screen.queryByText('purple butter')).toBeInTheDocument()
  
})