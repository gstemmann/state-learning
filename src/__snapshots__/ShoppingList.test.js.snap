// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Shopping List
    </h1>
    <form>
      <label
        for="name"
      >
        Product Name:
      </label>
      <input
        id="name"
        name="name"
        placeholder="Product Name"
        type="text"
        value=""
      />
      <label
        for="qty"
      >
        Number of Items:
      </label>
      <input
        id="qty"
        max="10"
        min="1"
        name="qty"
        placeholder="Product Quantity"
        type="range"
        value=""
      />
      <button
        type="submit"
      >
        Add Item
      </button>
    </form>
    <ul>
      <li>
        <ul>
          <li>
            Product Name:  
          </li>
          <li>
             Number of Items:  
          </li>
        </ul>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
